### PR TITLE
Homework5: Vladimir Natsevsky

### PR DESCRIPTION
**Pull request by JunglistMNSQ**

### CODE REVIEW

File: src/homework5/runner.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import allfunc
```
Comment (belovmd) [inline src/homework5/runner.py:1]: missed task description

File: src/homework5/runner.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+
+def run_fnc(*args):
```
Comment (belovmd) [inline src/homework5/runner.py]: wrong name according to task description

File: src/homework5/dec.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+(за все время вызовов, не только текущий запуск программы)
+
+"""
+
+
+def save_res(fnct):
+    def wrapper(*args, **kwargs):
+        res = fnct(*args, **kwargs)
+        with open('result.txt', 'a') as r:
+            r.write(res)
```
Comment (belovmd) [inline src/homework5/dec.py]: we can also add func name and datetime

File: src/homework5/00000001.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+
+def fnd_div(i):
+    a = i
+    count = 0
+    while a:
+        a = a >> 1
+        count += 1
+    div = 2 << count
+    while i % div:
```
Comment (belovmd) [inline src/homework5/00000001.py]: can it be resolved with &?

File: src/homework5/00000001.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+"""Вводится число найти его максимальный делитель, являющийся степенью
```
Comment (belovmd) [inline src/homework5/00000001.py:1]: nice file naming :)

**Review by belovmd — CHANGES_REQUESTED** [review]
- please remove rating file from PR
- please address next issues

**Review by belovmd — COMMENTED** [review]
please address previous review issues

File: src/homework5/runner.py (hunk 1 above)
Comment (JunglistMNSQ) [reply src/homework5/runner.py:1]: fixed

File: src/homework5/runner.py (hunk 2 above)
Comment (JunglistMNSQ) [reply src/homework5/runner.py]: fixed

File: src/homework5/00000001.py (hunk 4 above)
Comment (JunglistMNSQ) [reply src/homework5/00000001.py]: have done

File: src/homework5/dec.py (hunk 3 above)
Comment (JunglistMNSQ) [reply src/homework5/dec.py]: added

File: src/homework5/00000001.py (hunk 5 above)
Comment (JunglistMNSQ) [reply src/homework5/00000001.py:1]: i think the same

**Comment by JunglistMNSQ** [comment]
Fixed all issues.

File: src/homework5/00000001.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+двойки. 10(2) 16(16), 12(4)."""
+
+
+def fnd_div(i):
+    a = i
+    b = 2
+    res = 0
+    while not res:
+        res = a & b
+        b *= 2
```
Comment (belovmd) [inline src/homework5/00000001.py:11]: b<<2

**Review by belovmd — COMMENTED** [review]
9

**Pull request closed without merging**